core/include/thread_flags_group.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2025 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    core_thread_flags_group Thread Flags Group
+ * @ingroup     core
+ * @brief       Waiter groups for thread flags
+ *
+ *
+ * This API is optional and must be enabled by adding "core_thread_flags_group"
+ * to USEMODULE.
+ *
+ * A thread flags group allows setting thread flags for an arbitrary number of
+ * threads (called waiters) at the same time. The waiters can join and leave
+ * the group at any time. An additional advantage is that the signaler (the
+ * "flags setter") is de-coupled from the list of waiters, i.e. it does not
+ * need to know which specific thread must be signaled.
+ *
+ * Example (waiter):
+ *
+ *    static tfg_t group = THREAD_FLAGS_GROUP_INIT;
+ *
+ *    ...
+ *
+ *    thread_flags_group_join(&group);
+ *
+ *    while (!some_condition_is_met()) {
+ *        thread_flags_wait_any(SOME_FLAG);
+ *    }
+ *
+ *    thread_flags_group_leave(&group);
+ *
+ * Example (signaler):
+ *
+ *    fulfill_some_condition();
+ *    thread_flags_group_set(&group, SOME_FLAG);
+ *
+ * @{
+ *
+ * @file
+ * @brief       Thread Flags Group API
+ *
+ * @author      Mihai Renea <mihai.renea@ml-pa.com>
+ */
+
+#pragma once
+
+#include <limits.h>
+
+#include "atomic_utils.h"
+#include "thread.h"
+#include "thread_flags.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Some compilers don't define UINT_WIDTH */
+#ifndef UINT_WIDTH
```

Comment:
The defines are actually provided on the GCC 13.2.1 packed in Ubuntu, but guarded with:

```
#if (defined __STDC_WANT_IEC_60559_BFP_EXT__ \
     || (defined (__STDC_VERSION__) && __STDC_VERSION__ > 201710L))
```

This matches what https://en.cppreference.com/w/c/header/limits says.

```suggestion
/* UINT_WIDTH is only provided starting with -std=c23 or newer. Until RIOT
 * requires C23 as C version, we need provide it by hand when missing. */
#ifndef UINT_WIDTH
```